cmake/partition_manager.cmake

```diff
@@ -112,8 +112,8 @@ endforeach()
 # Explicitly add the dynamic partition image
 list(APPEND prefixed_images "${domain}:${dynamic_partition}")
 list(APPEND images ${dynamic_partition})
-list(APPEND input_files ${CMAKE_BINARY_DIR}/${generated_path}/pm.yml)
-list(APPEND header_files ${CMAKE_BINARY_DIR}/${generated_path}/pm_config.h)
+list(APPEND input_files ${CMAKE_CURRENT_BINARY_DIR}/${generated_path}/pm.yml)
+list(APPEND header_files ${CMAKE_CURRENT_BINARY_DIR}/${generated_path}/pm_config.h)
```

Comment:
In this case the `CMAKE_BINARY_DIR` and `CMAKE_CURRENT_BINARY_DIR` are identical, because the code is executed as part of boilerplate, but else the `CMAKE_CURRENT_BINARY_DIR` will change along with `add_subdirectory` calls, whereas `CMAKE_BINARY_DIR` is fixed.

In this particular case, where we refer to `pm` files, I think the best is to have a fixed path based on either `CMAKE_BINARY_DIR` or maybe even better `ZEPHYR_BINARY_DIR`.

Cause the `pm` files might be created in a `CMakeLists.txt` file that was included using `add_subdirectory` and then `CMAKE_CURRENT_BINARY_DIR` in this file and the `CMAKE_CURRENT_BINARY_DIR` value in the other `CMAKE_CURRENT_BINARY_DIR` could be different.

Whereas `ZEPHYR_BINARY_DIR` will be fixed in the whole build system.

As we also need to refer to `pm` in child images, then I suggest we share the `ZEPHYR_BINARY_DIR` variable from child image to parent, so that we can refer to it as `<image>_ZEPHYR_BINARY_DIR`, for example `spm_ZEPHYR_BINARY_DIR`, `b0_ZEPHYR_BINARY_DIR` and so on.

That way we can minize the risk of pointing to two different places in this code, and in child images.

Reply:
Great proposal, see if I was able to make it the way you wanted to :)